src/static_analysis/code.rs

```diff
@@ -75,7 +75,7 @@ pub fn code_analysis(manifest: Option<Manifest>, config: &Config, results: &mut
                         Some(f) => {
                             if let Err(e) =
                                    analyze_file(f.path(),
-                                                PathBuf::from(thread_dist_folder.as_str()),
+                                                (*thread_dist_folder).clone(),
```

Comment:
Not sure if this clone is needed. We could use both as `Path`s, by using `&f.path()` in the first parameter and `&thread_dist_folder` in the second one. Not sure, though, but it would be great if we could avoid the second allocation.